packages/@o3r/rules-engine/src/stores/rulesets/rulesets.selectors.ts

```diff
@@ -62,13 +55,22 @@ export const selectActiveRuleSets = createSelector(
     if (from) {
       return from.getTime() <= time;
     }
-
     return to && to.getTime() >= time;
-  }).map((ruleSet: Ruleset) => ruleSet.id));
+  })
+);
 
-/*
+/**
+ * Returns the rulesets ids which are not onDemand and in the validity range
+ */
+export const selectActiveRuleSets = createSelector(
+  selectRuleSetsInRange,
+  (ruleSets) => ruleSets
+    .filter((ruleSet: Ruleset) => (!(ruleSet.linkedComponents?.or?.length || ruleSet.linkedComponent)))
+    .map((ruleSet: Ruleset) => ruleSet.id));
+
+/**
  * Assign rulesetId to a component
- * @deprecated; It will be rmeoved in v12 with the selector using it
+ * @deprecated It will be removed in v12 with the selector using it
```

Comment:
this function is not exported, why can't we just remove it?

Reply:
because it is used in `selectRuleSetLinkComponents` which is exported